src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -134,6 +136,99 @@ def ensure_app_roles_is_a_set(
         return application_roles
 
 
+class ReleaseChannel(dict[str, Any]):
+    """
+    Represents a release channel.
+
+    This class is a dictionary with the following keys:
+    - name: The name of the release channel.
+    - description: The description of the release channel.
+    - created_on: The timestamp when the release channel was created.
+    - updated_on: The timestamp when the release channel was last updated.
+    - targets: The target accounts for the release channel.
+    - versions: The versions added to the release channel.
+    """
+
+    def __init__(self, data: dict[str, Any]):
+        targets = json.loads(str(data.get("targets"))) if data.get("targets") else {}
+        versions = json.loads(str(data.get("versions"))) if data.get("versions") else []
+
+        super().__init__(
+            {
+                "name": data.get("name") or "",
+                "description": data.get("description") or "",
+                "created_on": data.get("created_on"),
+                "updated_on": data.get("updated_on"),
+                "targets": targets,
+                "versions": versions,
+            }
+        )
+
+    @property
+    def name(self) -> str:
+        return self.get("name")  # type: ignore
+
+    @property
+    def description(self) -> str:
+        return self.get("description")  # type: ignore
+
+    @property
+    def created_on(self) -> Optional[datetime]:
+        return self.get("created_on")  # type: ignore
+
+    @property
+    def updated_on(self) -> Optional[datetime]:
+        return self.get("updated_on")  # type: ignore
+
+    @property
+    def targets(self) -> dict[str, Any]:
+        return self.get("targets")  # type: ignore
+
+    @property
+    def versions(self) -> list[str]:
+        return self.get("versions")  # type: ignore
+
+    def print_to_console(self, console: AbstractConsole) -> None:
+        """
+        Prints the release channel details to the console.
+        """
+        console.message(f"""[bold]{self.name}[/bold]""")
+        accounts_list: Optional[list[str]] = self.targets.get("accounts")
+        target_accounts = (
+            f"({', '.join(accounts_list)})"
+            if accounts_list is not None
+            else "ALL ACCOUNTS"
+        )
+
+        formatted_created_on = (
+            self.created_on.astimezone().strftime("%Y-%m-%d %H:%M:%S.%f %Z")
```

Comment:
nit:
Why this specific format? Do we already print some datetimes elsewhere?

Reply:
From the design doc, Bruno wanted something more human readable for console output - in this case, we are outputting the current timezone. 
Also, this is not set in stone, as it's only a console output. `--format=json` part is printed using the default.